src/app/carbon-estimation/carbon-estimation.component.ts

```diff
@@ -48,12 +43,20 @@ export class CarbonEstimationComponent implements OnInit, OnDestroy {
 
   constructor(private changeDetectorRef: ChangeDetectorRef) {
     effect(() => {
-      this.emissions = this.getOverallEmissionPercentages(this.carbonEstimation());
-      this.emissionAriaLabel = this.getAriaLabel(this.emissions);
+      if (this.showPlaceholderDiagram) {
+        this.chartData = placeholderData;
+        this.chartOptions = getChartOptions(true);
+        this.emissionAriaLabel = 'Placeholder for estimator of emissions';
+      } else {
+        this.chartData = this.getOverallEmissionPercentages(this.carbonEstimation()!);
+        this.chartOptions = getChartOptions(false);
+        this.emissionAriaLabel = this.getAriaLabel(this.chartData);
+      }
     });
   }
 
   public ngOnInit(): void {
+    this.chartOptions = getChartOptions(this.carbonEstimation() === null);
```

Comment:
Should this section be part of the effect above? Given it sets the chartOptions to a fresh object whenever the input carbonEstimation changes.

Reply:
I think this line should perhaps go in the constructor (and the argument to `getChartOptions` needs to be updated). I think it could go either there on in `ngOnInit` but perhaps the constructor is a more natural place to initialise it and it let's you drop the `!` in the variable declaration.

I'm not sure why but it seemed like setting the chart height when the chartOptions were overwritten in the effect wasn't required. It seemed to work fine when resizing and swapping between the estimation and placeholder. I'll look into it a bit more and see if I can work out what's going on. Perhaps it's still worth setting the chart height in the effect just in case.

Reply:
Nvm, it can't go in the constructor because the input is not available there :^) (It just looked like it worked because the initial value passed is currently `undefined`.)

Reply:
I've moved the chart configuration logic to a separate method which is called in `ngOnInit` and when the input is changed.

Reply:
I think we could still remove the call in `ngOnInit()` but we need to initialise the `chartOptions` on construction ie.

`public chartOptions: ChartOptions = getChartOptions(true);`

Without that, you end up with an error in the template that must prevent the signal/effect setup, meaning it never gets changes passed through.
```
TypeError: Cannot read properties of undefined (reading 'chart')
    at CarbonEstimationComponent_Template (carbon-estimation.component.html:28:27)
```
Bit of an optional one, you're either specifying the `chartOptions` in two places or calling `configureChart()` in two places.

Reply:
Are you seeing the error in the current setup or is that something that would need to be handled with the proposed change?

I think we probably need some chart configuration in `ngOnInit` because it depends on the `extraHeight` and `carbonEstimation` signals and I don't think they're available during the initial construction. I think it's going to be easier to do all of the initial setup in one place rather than split it between construction and `ngOnInit` and we can't configure the height without the `extraHeight` signal (it's used in `getChartHeight` which is used in `configureChart`)*.

Also, if we make a choice to set the chart options to the placeholder options on construction rather than basing it off the input we'd have a setup where sometimes the options are based on the input and sometimes they are a based on a choice we hard coded into the component and I think that's going to be potentially confusing when using the component. I think they should always depend on the input.

 *I think one thing we should look into refactoring is having the extraHeight be an explicit numerical argument that is passed into functions where it's used. At the moment the signal is accessed directly inside the `getChartHeight` function so it's not obvious when it's called that this function depends on a signal. I think it should also be a numerical input to the component rather than a string that is transformed inside the carbon-estimation component.

Reply:
Having said that, I think it would be nice to make it more explicit that `configureChart` is setting the `chartOptions`, it's not obvious that an assignment to the `chartOptions` variable is taking place when calling `configureChart`.

Reply:
Sorry for the confusion, I only see the error if I remove `this.configureChart(this.carbonEstimation());` from `ngOnInit()`, without also setting the `chartOptions` to something valid.

The effect still seems to get called (probably just after `ngOnInit()`) - so even if I initialise `chartOptions` to the non-placeholder options, it will update to use the grey one and you never actually see the colours. May be behaviour that we shouldn't rely on tho.

Reply:
Also if we want to use signals properly this should maybe use `computed()` instead of `effect()` but that's a bigger change.

Reply:
Ah, for some reason I thought the effect wasn't running when the component is initialised but it does indeed seem to be running after `ngOnInit`. Even though it looks like the initial setup will be immediately overwritten by the effect running I think it will still be clearer to do the initialisation in `ngOnInit` based on the inputs. As you alluded to, I think that will also be more robust if the behaviour of `effect()` changed in the future or we swapped to `computed()` and that didn't run after initialisation.